Feature/pt 50 mobile specialists search

KaterynaWalls opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added specialist ratings, review counts, and district information to specialist profiles.
  - Introduced a new Bio component for displaying specialist biographies with improved formatting and truncation options.

- **Enhancements**
  - Specialist cards now display ratings, review counts, and improved image handling with device-specific placeholders.
  - Pagination and page navigation are now URL-driven for improved accessibility and usability.
  - Skeleton loaders and state displays feature responsive grid layouts and adaptive animation sizes.
  - Improved responsive design and styling across specialist lists, cards, back button, and buttons.
  - Updated page layout and heading structure for better responsiveness on the search specialists page.

- **Bug Fixes**
  - Minor whitespace cleanup in specialist descriptions.

- **Style**
  - Updated CSS for better responsive behavior, button sizing, and font-face formatting.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

KaterynaWalls committed:
merge branch dev into feature/PT-50, work in progress mobile searchSpecialists

KaterynaWalls committed:
Merge branch 'feature/PT-50-mobile-specialists-search' of github.com:petHelpMarketplace/frontend into feature/PT-50-mobile-specialists-search

KaterynaWalls committed:
wip(PT-50): continue mobile layout implementation for specialist search

KaterynaWalls committed:
refactor(PT-50):  pagination logic, index.css styles & responsive (адаптація, hover, active, aria)

KaterynaWalls committed:
style: specialist card mobile layout on narrow screens

KaterynaWalls committed:
refactor(PT-50): refactor and adaptive pagination, update SearchSpecialistsPage

KaterynaWalls committed:
feat(pt-50): improved adaptive layout and placeholder images

KaterynaWalls committed:
fix(PT-50): clamp currentPage to [1, totalPages] to prevent invalid page param in URL (pagination)

Sashko-kapillar reviewed (approved):
it`s fine

yuliia-shl commented on src/features/searchSpecialists/components/SpecialistCard.tsx:
```diff
@@ -27,86 +31,143 @@ const SpecialistCard = ({ specialist }: Props) => {
+          <div className="flex items-center">
+            <span className="inline-flex items-center gap-2 whitespace-nowrap">
+              <h2 className="text-lg font-semibold text-fire leading-[122%] xl:leading-[150%]">
+                {name} {shortFamilyName}
```
Вкладення ```<h2>``` у ```<span>```: Неправильно з точки зору HTML: ```<h2>``` - це блоковий елемент, а ```<span>``` - стрічковий.

KaterynaWalls committed:
Merge remote-tracking branch 'origin' into feature/PT-50-mobile-specialists-search

KaterynaWalls committed:
refactor(PT-50): fix spacing below pagination, improve SpecialistCard markup

- Adjust pagination bottom margin: mb-10 (mobile 40px), xl:mb-[58px] (desktop 58px) to match design
- Remove heading tag (h2) from SpecialistCard; replaced with plain <div> for specialist's name as per best practice
- Remove unnecessary <span> wrapper around name in SpecialistCard

KaterynaWalls committed:
fix(PT-50): improve image error handling for responsive placeholders

yuliia-shl commented on src/index.css:
```diff
@@ -92,7 +92,7 @@ button:not(:disabled) {
 @layer components {
   .btn {
-    @apply bg-tenn text-alabaster rounded-2xl px-5 min-w-[200px] h-[40px] transition-shadow duration-300 ease-in-out;
+    @apply bg-tenn text-alabaster max-w-[158px] h-[35px] rounded-2xl xl:px-5 xl:min-w-[200px] xl:h-[40px] transition-shadow duration-300 ease-in-out;
```
Треба виправити кнопки
![image](https://github.com/user-attachments/assets/9a64e9fe-519e-4012-8f01-8984caacb4cd)

KaterynaWalls committed:
fix(PT-50): fix btn

KaterynaWalls committed:
fix(PT-50): fix btn
